applications/OptimizationApplication/python_scripts/controls/thickness/shell_thickness_control.py

```diff
@@ -0,0 +1,235 @@
+        self.model = model
+        self.optimization_problem = optimization_problem
+
+        default_settings = Kratos.Parameters("""{
+            "controlled_model_part_names": [],
+            "filter_settings": {
+                "type" : "implicit",
+                "radius": 0.000000000001,
+                "linear_solver_settings" : {}
+            },
+            "beta_settings": {
+                "initial_value" : 25,
+                "max_value" : 25,
+                "adaptive" : false,
+                "increase_fac" : 1.5,
+                "update_period" : 20
+            },
+            "SIMP_power_fac": 3,
+            "initial_thickness":0.000001,
+            "physical_thicknesses": [],
+            "fixed_model_parts": [],
+            "fixed_model_parts_thicknesses": [],
+            "utilities": []
+        }""")
+
+        self.parameters.ValidateAndAssignDefaults(default_settings)
+        self.parameters["beta_settings"].ValidateAndAssignDefaults(default_settings["beta_settings"])
+        self.parameters["filter_settings"].ValidateAndAssignDefaults(default_settings["filter_settings"])
+
+        self.penal_factor = self.parameters["SIMP_power_fac"].GetInt()
+        self.initial_beta = self.parameters["beta_settings"]["initial_value"].GetDouble()
+        self.physical_thicknesses = self.parameters["physical_thicknesses"].GetVector()
+        num_phys_thick = len(self.physical_thicknesses)
+        if num_phys_thick == 0:
+            raise RuntimeError(f"The  physical_thicknesses can not be empty, at least min and max should be provided.")
+        self.filtered_thicknesses = [i for i in range(len(self.physical_thicknesses))]
+
+        self.filter_type = self.parameters["filter_settings"]["type"].GetString()
+        self.supported_filter_types = ["implicit"]
+
+        if not self.filter_type in self.supported_filter_types:
+            raise RuntimeError(f"The specified filter type \"{self.filter_type}\" is not supported. Followings are the variables:\n\t" + "\n\t".join([k for k in self.supported_filter_types]))
+
+        controlled_model_parts = [model[model_part_name] for model_part_name in parameters["controlled_model_part_names"].GetStringArray()]
+        if len(controlled_model_parts) == 0:
+            raise RuntimeError(f"No model parts were provided for ShellThicknessControl. [ control name = \"{self.GetName()}\"]")
+
+        self.model_part = ModelPartUtilities.GetOperatingModelPart(ModelPartUtilities.OperationType.UNION, f"control_{self.GetName()}", controlled_model_parts, False)
+
+        self.fixed_model_parts = self.parameters["fixed_model_parts"].GetStringArray()
+        self.fixed_model_parts_thicknesses = self.parameters["fixed_model_parts_thicknesses"].GetVector()
+        if len(self.fixed_model_parts) != len(self.fixed_model_parts_thicknesses):
+            raise RuntimeError(f"fixed_model_parts and fixed_model_parts_thicknesses should have the same size !")
```

Comment:
These checks will be redundant if a dictionary is used for these settings, and I think it is more readable from the user persepective and developer perspective.